subprojects/dependency-management/src/main/java/org/gradle/internal/locking/LockFileReaderWriter.java

```diff
@@ -41,9 +42,11 @@
                                                  "# This file is expected to be part of source control.\n";
 
     private final Path lockFilesRoot;
+    private final DomainObjectContext context;
     private final boolean isProjectScope;
 
-    public LockFileReaderWriter(FileResolver fileResolver, boolean isProjectScope) {
+    public LockFileReaderWriter(FileResolver fileResolver, DomainObjectContext context, boolean isProjectScope) {
+        this.context = context;
         this.isProjectScope = isProjectScope;
```

Comment:
Again, I'd like to see if we can use `this.isProjectScope = !context.isScript()` or `this.isProjectScope = (context.getProjectPath() == null)`